src/store/registration/saga.ts

```diff
@@ -140,4 +143,21 @@ export function* saga() {
     const action = yield take(SagaActionTypes.UpdateProfile);
     success = yield call(updateProfile, action);
   } while (!success);
+
+  yield spawn(conversationsWatcher);
+}
+
+export function* channelsLoaded() {
+  const existingConversationsList = yield select(rawConversationsList());
+  if (existingConversationsList.length > 0) {
+    yield put(setActiveMessengerId(existingConversationsList[0]));
+  }
+}
+
+function* conversationsWatcher() {
+  const channel = yield call(conversationsChannel);
+  const payload = yield take(channel, '*');
+  if (payload.loaded) {
+    yield call(channelsLoaded);
+  }
 }
```

Comment:
to clarify, `yield take(channel, '*')` will "listen" for all events were the "channel" redux state is updated, and is checking if `{ loaded: true }` is set? (after which it sets the active messenger id)

Reply:
Not really the "redux state". It's events that come across that specific channel. In this case, it's the conversationsChannel that we're listening to.

The `{loaded: true}` is just the value I chose to use when publishing the event. I decided this conversation event was a "conversations loaded" event and this seemed an ok way to indication that. Another option would be to publish it more like an action payload: `{type: "CONVERSATIONS_LOADED"}`. Maybe that would make more sense?